File: pythonlibrary/aibrd/models/brd.py
from typing import Literal, Optional
from pydantic import BaseModel

FileType = Literal["pdf", "docx", "markdown"]


class RawBRD(BaseModel):
    text: str
    source: str
    file_type: FileType


class BRDChunk(BaseModel):
    text: str
    index: int
    total: int
    module_hint: Optional[str] = None

File: vscodebase/src/core/models/brd.ts
export type FileType = 'pdf' | 'docx' | 'markdown'

export interface RawBRD {
  text: string
  source: string
  fileType: FileType
}

export interface BRDChunk {
  text: string
  index: number
  total: number
  moduleHint?: string
}
